airsonic-main/src/main/java/org/airsonic/player/dao/DaoHelper.java

```diff
@@ -46,4 +46,14 @@
     NamedParameterJdbcTemplate getNamedParameterJdbcTemplate();
 
     DataSource getDataSource();
+
+    /**
+     * Tries to perform a checkpoint against the database, if supported
+     *
+     * Database checkpoints will make sure that the database is written on the disk
+     * and optimize on-disk storage.
+     *
+     * @return true if the checkpoint succeeded, false otherwise
+     */
+    public default boolean checkpoint() { return false; }
```

Comment:
It dones't look like the return value is ever checked. I'd just leave the return value `void` - probably better to throw an exception if it fails anyways.